File: home-hub/backend/api/schemas/sonos.py
"""
Pydantic schemas for Sonos API requests and responses.
"""
from typing import Optional

from pydantic import BaseModel, Field


class SonosStatus(BaseModel):
    """Current Sonos playback status."""

    state: str
    track: str = ""
    artist: str = ""
    album: str = ""
    art_url: str = ""
    duration: str = "0:00:00"
    position: str = "0:00:00"
    volume: int = 0
    mute: bool = False


class VolumeRequest(BaseModel):
    """Request to set Sonos volume."""

    volume: int = Field(ge=0, le=100)


class TTSRequest(BaseModel):
    """Request to speak text on Sonos."""

    text: str = Field(min_length=1, max_length=500)
    volume: Optional[int] = Field(None, ge=0, le=100)

File: frontend-svelte/src/lib/stores/sonos.js
import { writable } from 'svelte/store'

/**
 * @typedef {Object} SonosState
 * @property {string} state — "PLAYING" | "PAUSED_PLAYBACK" | "STOPPED" | ...
 * @property {string} track
 * @property {string} artist
 * @property {string} album
 * @property {string} art_url
 * @property {number} volume
 * @property {boolean} mute
 */

/** @type {SonosState} */
const initial = {
  state: 'STOPPED',
  track: '',
  artist: '',
  album: '',
  art_url: '',
  volume: 0,
  mute: false,
}

/** @type {import('svelte/store').Writable<SonosState>} */
export const sonos = writable(initial)
